src/Build/Evaluation/ProjectRootElementCache.cs

```diff
@@ -275,7 +275,13 @@ private bool IsInvalidEntry(string projectFile, ProjectRootElement projectRootEl
 
                 projectRootElement = openProjectRootElement(projectFile, this);
                 ErrorUtilities.VerifyThrowInternalNull(projectRootElement, "projectRootElement");
-                ErrorUtilities.VerifyThrow(projectRootElement.FullPath == projectFile, "Got project back with incorrect path");
+                ErrorUtilities.VerifyThrow(
```

Comment:
This line basically checks if `projectFile` that came to this function is indeed a full path, as it should be.